integration/util.go

```diff
@@ -17,3 +21,53 @@ func checksumOf(bytes []byte) string {
 	hash.Write(bytes)
 	return hex.EncodeToString(hash.Sum(nil))
 }
+
+func listArchiveContents(path string) ([]string, error) {
+	output, err := command.NewFactory(env.NewRepository()).
+		Create("tar", []string{"-tf", path}, nil).
+		RunAndReturnTrimmedOutput()
+
+	if err != nil {
+		return nil, err
+	}
+
+	return strings.Split(output, "\n"), nil
+}
+
+func checkTools() {
+	_, err := exec.LookPath("zstd")
+	if err != nil {
+		panic("zstd is required for integration tests")
+	}
+}
+
+type fakeEnvRepo struct {
```

Comment:
I am happy to see other test doubles than just mocks. But I also need to ask why did not you just use mockery to generate a mock for the env repo object?

I am only asking because once I faced the same choice and because I only saw mocks being used in the tests I also went down that route. 😄

Reply:
I was curious how hard it would be to create fakes, and I was also too lazy to set up Mockery and find out the exact commands to generate the mocks. And I liked the end result, the `EnvRepository` is such a simple interface that it can be faked very easily.

Reply:
At my previous job I was working on an internal video playback framework. This meant that we also had to have an additional testing package for our consumers so they can easily write tests for their stuff. This testing package had stubs, spys, fakes and dummies in it. So I am all in favour of not only using mocks in our tests.